core/app/models/spree/store_credit.rb

```diff
@@ -288,4 +289,11 @@ def associate_credit_type
       self.credit_type = Spree::StoreCreditType.find_by_name(credit_type_name)
     end
   end
+
+  def apply_to_orders_in_confirm_state
```

Comment:
Is there another way to do this? This makes too many assumptions about the order state machine and how it works.

The store credit class should have no knowledge of what states an order has or what opperations happen on transitions between them.